datafusion/functions-nested/src/set_ops.rs

```diff
@@ -513,9 +513,6 @@ fn general_array_distinct<OffsetSize: OffsetSizeTrait>(
     array: &GenericListArray<OffsetSize>,
     field: &FieldRef,
 ) -> Result<ArrayRef> {
-    if array.len() == 0 {
```

Comment:
do we really need to remove this check? 🤔 it prevents unnecessary work/allocations if nothing is in the `array`